### PR TITLE
Clarify reference frames.

### PR DESCRIPTION
**Pull request by eggrobin**
- Avoid the sloppy JPL explanation based on dynamical properties;
- the ICRS has coordinates consistent with FK5 J2000 equatorial coordinates, there is no ICRS ecliptic, see [Capitaine and Soffel 2015, On the definition and use of the ecliptic in modern astronomy](https://arxiv.org/pdf/1501.05534.pdf); rename `ICRFJ2000Equator` to `ICRS` and get rid of `ICRFJ2000Ecliptic`.
- introduce the ITRS in preparation for tests based on ILRS products;
- introduce the TIRS, CIRS, and GCRS to enable conversions from the ITRS to the ICRS;
- document the errors arising from not versioning the ITRFs; this is be fine for our purposes for the recent ones;
- document the errors arising from identifying the ITRS and TIRS, and the CIRS and GCRS; these are likely not tolerable.

### CODE REVIEW

File: astronomy/frames.hpp [hunk 1]
```diff
@@ -19,57 +19,131 @@ using quantities::si::ArcMinute;
...
+// The x axis is close to the dynamical equinox at J2000.
+// The z axis is perpendicular to the xy plane and points to the northern side
+// of the xy plane.
+// To be honest:
+// - There are several realizations of the ICRS, namely ICRF (ICRF1) and its
+//   extensions, ICRF2, and the optical Hipparcos Celestial Reference Frame
+//   (HCRF); these realizations have different errors, but are consistent. ICRF3
+//   is in preparation, as well as an optical realization based on Gaia data.
+// - The terms Barycentric Celestial Reference System (BCRS) and ICRS are often
+//   used interchangeably; the relativistic framework, including the metric
```
Comment (pleroy) [inline astronomy/frames.hpp]: Should we have a typename for BCRS?

File: astronomy/frames.hpp [hunk 2]
```diff
@@ -19,57 +19,131 @@ using quantities::si::ArcMinute;
...
-          serialization::Frame::ICRF_J2000_EQUATOR, true>;
+// The Geocentric Celestial Reference System.
+// The origin is the centre of mass of the whole Earth system, including oceans
+// and atmosphere; the axes are those of the ICRS.
+// To be honest:
+// - The BCRS and GCRS are relativistic reference systems.
+// - The transformation between BCRS and GCRS spatial coordinates has no
+//   rotational component, i.e., the GCRS is kinematically non-rotating with
+//   respect to the BCRS; as a result, it rotates dynamically with respect to
+//   the BCRS; in particular, the equations of motion in the GCRS include de
```
Comment (pleroy) [inline astronomy/frames.hpp]: I don't understand.  Does it rotate or not?

File: astronomy/frames.hpp [hunk 3]
```diff
@@ -19,57 +19,131 @@ using quantities::si::ArcMinute;
...
+// To be honest:
+// - There are many realizations of the ITRS; as of this writing, ICRFs 89, 90,
+//   91, 92, 93, 94, 96, 97, 2000, 2005, 2008, and 2014.  They are related by
+//   linearly time-dependent linearized similarity transformations with nonzero
+//   scaling, which the current abstractions of Principia cannot support; see
+//   Transformation parameters from ITRF2014 to past ITRFs,
+//   http://itrf.ensg.ign.fr/doc_ITRF/Transfo-ITRF2014_ITRFs.txt.  Identifying
+//   recent ITRFs (2005 and later) results in errors of a few millimetres at the
+//   geocentre, scale errors on the order of 1 part-per-billion, and angular
+//   errors below 1 μas.  Identifying earlier ITRFs results in errors may result
```
Comment (pleroy) [inline astronomy/frames.hpp]: Garbled.

File: astronomy/mercury_perihelion_test.cpp [hunk 4]
```diff
@@ -213,13 +211,12 @@ TEST_F(MercuryPerihelionTest, DISABLED_Year2050) {
   auto const& mercury_trajectory =
       solar_system_1950_.trajectory(*ephemeris_, "Mercury");
 
-  RelativeDegreesOfFreedom<ICRFJ2000Equator> const relative_degrees_of_freedom =
+  RelativeDegreesOfFreedom<ICRS> const relative_degrees_of_freedom =
       mercury_trajectory.EvaluateDegreesOfFreedom(t_2050_) -
       sun_trajectory.EvaluateDegreesOfFreedom(t_2050_);
-  KeplerOrbit<ICRFJ2000Equator> orbit(
+  KeplerOrbit<ICRS> orbit(
       *sun_, *mercury_, relative_degrees_of_freedom, t_2050_);
```
Comment (pleroy) [inline astronomy/mercury_perihelion_test.cpp]: Fits on the previous line?

File: astronomy/mercury_perihelion_test.cpp [hunk 5]
```diff
@@ -178,13 +177,12 @@ TEST_F(MercuryPerihelionTest, Year1960) {
   auto const& mercury_trajectory =
       solar_system_1950_.trajectory(*ephemeris_, "Mercury");
 
-  RelativeDegreesOfFreedom<ICRFJ2000Equator> const relative_degrees_of_freedom =
+  RelativeDegreesOfFreedom<ICRS> const relative_degrees_of_freedom =
       mercury_trajectory.EvaluateDegreesOfFreedom(t_1960_) -
       sun_trajectory.EvaluateDegreesOfFreedom(t_1960_);
-  KeplerOrbit<ICRFJ2000Equator> orbit(
+  KeplerOrbit<ICRS> orbit(
       *sun_, *mercury_, relative_degrees_of_freedom, t_1960_);
```
Comment (pleroy) [inline astronomy/mercury_perihelion_test.cpp]: Fits on the previous line?

File: astronomy/mercury_perihelion_test.cpp [hunk 6]
```diff
@@ -137,13 +137,12 @@ TEST_F(MercuryPerihelionTest, Year1950) {
   auto const& mercury_trajectory =
       solar_system_1950_.trajectory(*ephemeris_, "Mercury");
 
-  RelativeDegreesOfFreedom<ICRFJ2000Equator> const relative_degrees_of_freedom =
+  RelativeDegreesOfFreedom<ICRS> const relative_degrees_of_freedom =
       mercury_trajectory.EvaluateDegreesOfFreedom(t_1950_) -
       sun_trajectory.EvaluateDegreesOfFreedom(t_1950_);
-  KeplerOrbit<ICRFJ2000Equator> orbit(
+  KeplerOrbit<ICRS> orbit(
       *sun_, *mercury_, relative_degrees_of_freedom, t_1950_);
```
Comment (pleroy) [inline astronomy/mercury_perihelion_test.cpp]: Fits on the previous line?

File: astronomy/solar_system_dynamics_test.cpp [hunk 7]
```diff
@@ -110,9 +110,9 @@ class SolarSystemDynamicsTest : public ::testing::Test {
 
   OrbitError CompareOrbits(
       int const index,
```
Comment (pleroy) [inline astronomy/solar_system_dynamics_test.cpp]: Fits on the previous line?

File: astronomy/solar_system_dynamics_test.cpp [hunk 8]
```diff
@@ -121,12 +121,12 @@ class SolarSystemDynamicsTest : public ::testing::Test {
         SolarSystemFactory::name(SolarSystemFactory::parent(index));
     auto const body = system.massive_body(ephemeris, name);
     auto const parent =
-        dynamic_cast_not_null<RotatingBody<ICRFJ2000Equator> const*>(
+        dynamic_cast_not_null<RotatingBody<ICRS> const*>(
```
Comment (pleroy) [inline astronomy/solar_system_dynamics_test.cpp]: Fits on the previous line?

File: astronomy/solar_system_dynamics_test.cpp [hunk 9]
```diff
@@ -197,22 +194,22 @@ class SolarSystemDynamicsTest : public ::testing::Test {
           OrientedAngleBetween(z, solar_system_angular_momentum, normal);
       EXPECT_THAT(declination_of_invariable_plane, Gt(0 * Radian));
       rotation =
-          Rotation<ICRFJ2000Equator, ParentEquator>(
+          Rotation<ICRS, ParentEquator>(
```
Comment (pleroy) [inline astronomy/solar_system_dynamics_test.cpp]: Fits on the previous line?

File: astronomy/solar_system_dynamics_test.cpp [hunk 10]
```diff
@@ -197,22 +194,22 @@ class SolarSystemDynamicsTest : public ::testing::Test {
...
               declination_of_invariable_plane,
               normal,
               DefinesFrame<ParentEquator>{});
     } else {
       auto const ω = parent->angular_velocity();
-      Bivector<double, ICRFJ2000Equator> const normal =
+      Bivector<double, ICRS> const normal =
           Commutator(z, Normalize(ω));
```
Comment (pleroy) [inline astronomy/solar_system_dynamics_test.cpp]: Fits on the previous line?

File: astronomy/solar_system_dynamics_test.cpp [hunk 11]
```diff
@@ -197,22 +194,22 @@ class SolarSystemDynamicsTest : public ::testing::Test {
...
       EXPECT_THAT(parent_axis_declination, Gt(0 * Radian));
-      rotation = Rotation<ICRFJ2000Equator, ParentEquator>(
+      rotation = Rotation<ICRS, ParentEquator>(
           parent_axis_declination, normal, DefinesFrame<ParentEquator>{});
     }
-    RigidMotion<ICRFJ2000Equator, ParentEquator> const
+    RigidMotion<ICRS, ParentEquator> const
         to_parent_equator(
-            {ICRFJ2000Equator::origin,
+            {ICRS::origin,
```
Comment (pleroy) [inline astronomy/solar_system_dynamics_test.cpp]: Fits on the previous line?

File: astronomy/молния_orbit_test.cpp [hunk 12]
```diff
@@ -72,27 +72,27 @@ class МолнияOrbitTest : public ::testing::Test {
...
         "sol_initial_state_jd_2451545_000000000.proto.txt");
-std::unique_ptr<Ephemeris<ICRFJ2000Equator>> МолнияOrbitTest::ephemeris_;
+std::unique_ptr<Ephemeris<ICRS>> МолнияOrbitTest::ephemeris_;
 
 #if !defined(_DEBUG)
 
 TEST_F(МолнияOrbitTest, Satellite) {
   auto const earth_body =
-      dynamic_cast_not_null<OblateBody<ICRFJ2000Equator> const*>(
+      dynamic_cast_not_null<OblateBody<ICRS> const*>(
```
Comment (pleroy) [inline astronomy/молния_orbit_test.cpp]: Fits on the previous line?

File: benchmarks/ephemeris.cpp [hunk 13]
```diff
@@ -210,29 +213,40 @@ void EphemerisL4ProbeBenchmark(SolarSystemFactory::Accuracy const accuracy,
...
+    Velocity<Ecliptic> const sun_earth_velocity = equatorial_to_ecliptic(
+        from_barycentric(earth_degrees_of_freedom.velocity() -
+                         sun_degrees_of_freedom.velocity()));
+    Velocity<Ecliptic> const sun_l4_velocity = l4_rotation(sun_earth_velocity);
     trajectory.Append(at_спутник_1_launch->epoch(),
                       DegreesOfFreedom<Barycentric>(
                           sun_degrees_of_freedom.position() +
                               to_barycentric(
-                                  ICRFJ200EquatorialToEcliptic.Inverse()(
+                                  ecliptic_to_equatorial(
```
Comment (pleroy) [inline benchmarks/ephemeris.cpp]: Fits on the previous line?

File: benchmarks/ephemeris.cpp [hunk 14]
```diff
@@ -210,29 +213,40 @@ void EphemerisL4ProbeBenchmark(SolarSystemFactory::Accuracy const accuracy,
...
                                       sun_l4_displacement)),
                           sun_degrees_of_freedom.velocity() +
                               to_barycentric(
-                                  ICRFJ200EquatorialToEcliptic.Inverse()(
+                                  ecliptic_to_equatorial(
```
Comment (pleroy) [inline benchmarks/ephemeris.cpp]: Fits on the previous line?

File: benchmarks/newhall.cpp [hunk 15]
```diff
@@ -67,31 +67,31 @@ void BM_NewhallApproximationDouble(benchmark::State& state) {
 template<typename Result,
          Result (*newhall)(
              int degree,
```
Comment (pleroy) [inline benchmarks/newhall.cpp]: Fits on the previous line?

File: ksp_plugin_test/plugin_test.cpp [hunk 16]
```diff
@@ -272,8 +272,8 @@ class PluginTest : public testing::Test {
         serialized.get());
   }
 
-  static RigidMotion<ICRFJ2000Equator, Barycentric> const id_icrf_barycentric_;
-  not_null<std::unique_ptr<SolarSystem<ICRFJ2000Equator>>> solar_system_;
+  static RigidMotion<ICRS, Barycentric> const id_icrf_barycentric_;
```
Comment (pleroy) [inline ksp_plugin_test/plugin_test.cpp]: ircs?

File: physics/barycentric_rotating_dynamic_frame_test.cpp [hunk 17]
```diff
@@ -106,36 +106,36 @@ class BarycentricRotatingDynamicFrameTest : public ::testing::Test {
...
-  StrictMock<MockEphemeris<ICRFJ2000Equator>> mock_ephemeris_;
+  DegreesOfFreedom<ICRS> const centre_of_mass_initial_state_;
+  StrictMock<MockEphemeris<ICRS>> mock_ephemeris_;
 
-  std::unique_ptr<BarycentricRotatingDynamicFrame<ICRFJ2000Equator, MockFrame>>
+  std::unique_ptr<BarycentricRotatingDynamicFrame<ICRS, MockFrame>>
       mock_frame_;
   std::unique_ptr<
-      BarycentricRotatingDynamicFrame<ICRFJ2000Equator, BigSmallFrame>>
+      BarycentricRotatingDynamicFrame<ICRS, BigSmallFrame>>
```
Comment (pleroy) [inline physics/barycentric_rotating_dynamic_frame_test.cpp]: Fits on the previous line?

File: physics/barycentric_rotating_dynamic_frame_test.cpp [hunk 18]
```diff
@@ -106,36 +106,36 @@ class BarycentricRotatingDynamicFrameTest : public ::testing::Test {
...
+  DegreesOfFreedom<ICRS> const small_initial_state_;
   GravitationalParameter const small_gravitational_parameter_;
-  DegreesOfFreedom<ICRFJ2000Equator> const centre_of_mass_initial_state_;
-  StrictMock<MockEphemeris<ICRFJ2000Equator>> mock_ephemeris_;
+  DegreesOfFreedom<ICRS> const centre_of_mass_initial_state_;
+  StrictMock<MockEphemeris<ICRS>> mock_ephemeris_;
 
-  std::unique_ptr<BarycentricRotatingDynamicFrame<ICRFJ2000Equator, MockFrame>>
+  std::unique_ptr<BarycentricRotatingDynamicFrame<ICRS, MockFrame>>
       mock_frame_;
```
Comment (pleroy) [inline physics/barycentric_rotating_dynamic_frame_test.cpp]: Fits on the previous line?

File: physics/barycentric_rotating_dynamic_frame_test.cpp [hunk 19]
```diff
@@ -219,26 +219,26 @@ TEST_F(BarycentricRotatingDynamicFrameTest, CoriolisAcceleration) {
        Velocity<MockFrame>({(80 - 30) * Metre / Second,
                             (-60 - 40) * Metre / Second,
                             0 * Metre / Second})};
-  DegreesOfFreedom<ICRFJ2000Equator> const big_dof =
-      {Displacement<ICRFJ2000Equator>({0.8 * Metre, -0.6 * Metre, 0 * Metre}) +
-           ICRFJ2000Equator::origin,
-       Velocity<ICRFJ2000Equator>({-16 * Metre / Second,
+  DegreesOfFreedom<ICRS> const big_dof =
+      {Displacement<ICRS>({0.8 * Metre, -0.6 * Metre, 0 * Metre}) +
+           ICRS::origin,
```
Comment (pleroy) [inline physics/barycentric_rotating_dynamic_frame_test.cpp]: Fits on the previous line?  (Many places.)

File: physics/barycentric_rotating_dynamic_frame_test.cpp [hunk 20]
```diff
@@ -219,26 +219,26 @@ TEST_F(BarycentricRotatingDynamicFrameTest, CoriolisAcceleration) {
...
+       Velocity<ICRS>({-16 * Metre / Second,
                                    12 * Metre / Second,
                                    0 * Metre / Second})};
-  DegreesOfFreedom<ICRFJ2000Equator> const small_dof =
-      {Displacement<ICRFJ2000Equator>({5 * Metre, 5 * Metre, 0 * Metre}) +
-           ICRFJ2000Equator::origin,
-       Velocity<ICRFJ2000Equator>({40 * Metre / Second,
+  DegreesOfFreedom<ICRS> const small_dof =
+      {Displacement<ICRS>({5 * Metre, 5 * Metre, 0 * Metre}) +
+           ICRS::origin,
```
Comment (pleroy) [inline physics/barycentric_rotating_dynamic_frame_test.cpp]: Fits on the previous line?

File: physics/barycentric_rotating_dynamic_frame_test.cpp [hunk 21]
```diff
@@ -219,26 +219,26 @@ TEST_F(BarycentricRotatingDynamicFrameTest, CoriolisAcceleration) {
...
+       Velocity<ICRS>({40 * Metre / Second,
```
Comment (pleroy) [inline physics/barycentric_rotating_dynamic_frame_test.cpp]: Bad indent.  (Many places.)

File: physics/barycentric_rotating_dynamic_frame_test.cpp [hunk 22]
```diff
@@ -219,26 +219,26 @@ TEST_F(BarycentricRotatingDynamicFrameTest, CoriolisAcceleration) {
...
-      Barycentre<DegreesOfFreedom<ICRFJ2000Equator>, GravitationalParameter>(
+  DegreesOfFreedom<ICRS> const barycentre_dof =
+      Barycentre<DegreesOfFreedom<ICRS>, GravitationalParameter>(
           {big_dof, small_dof},
           {big_gravitational_parameter_, small_gravitational_parameter_});
-  EXPECT_THAT(barycentre_dof.position() - ICRFJ2000Equator::origin,
-              Eq(Displacement<ICRFJ2000Equator>(
+  EXPECT_THAT(barycentre_dof.position() - ICRS::origin,
+              Eq(Displacement<ICRS>(
                      {2 * Metre, 1 * Metre, 0 * Metre})));
```
Comment (pleroy) [inline physics/barycentric_rotating_dynamic_frame_test.cpp]: Fits on the previous line?  (Many places.)

File: physics/body_centred_body_direction_dynamic_frame_test.cpp [hunk 23]
```diff
@@ -103,35 +103,35 @@ class BodyCentredBodyDirectionDynamicFrameTest : public ::testing::Test {
...
-  DegreesOfFreedom<ICRFJ2000Equator> small_initial_state_;
+  DegreesOfFreedom<ICRS> small_initial_state_;
   GravitationalParameter small_gravitational_parameter_;
-  StrictMock<MockEphemeris<ICRFJ2000Equator>> mock_ephemeris_;
+  StrictMock<MockEphemeris<ICRS>> mock_ephemeris_;
 
   std::unique_ptr<
-      BodyCentredBodyDirectionDynamicFrame<ICRFJ2000Equator, MockFrame>>
+      BodyCentredBodyDirectionDynamicFrame<ICRS, MockFrame>>
       mock_frame_;
```
Comment (pleroy) [inline physics/body_centred_body_direction_dynamic_frame_test.cpp]: Fits on the previous line?

File: physics/body_centred_body_direction_dynamic_frame_test.cpp [hunk 24]
```diff
@@ -103,35 +103,35 @@ class BodyCentredBodyDirectionDynamicFrameTest : public ::testing::Test {
...
   std::unique_ptr<
-      BodyCentredBodyDirectionDynamicFrame<ICRFJ2000Equator, BigSmallFrame>>
+      BodyCentredBodyDirectionDynamicFrame<ICRS, BigSmallFrame>>
```
Comment (pleroy) [inline physics/body_centred_body_direction_dynamic_frame_test.cpp]: Fits on the previous line?

File: physics/body_test.cpp [hunk 25]
```diff
@@ -269,7 +269,7 @@ TEST_F(BodyTest, AllFrames) {
   TestRotatingBody<serialization::Frame::SolarSystemTag,
                   serialization::Frame::ICRF_J2000_ECLIPTIC>();
   TestRotatingBody<serialization::Frame::SolarSystemTag,
-                  serialization::Frame::ICRF_J2000_EQUATOR>();
+                  serialization::Frame::ICRS>();
```
Comment (pleroy) [inline physics/body_test.cpp]: Bad indent (was there before).  Also 2 lines above.

File: physics/ephemeris_test.cpp [hunk 26]
```diff
@@ -132,30 +132,30 @@ class EphemerisTest
...
+        Displacement<ICRS>({0 * Metre,
                                         4e8 * Metre,
                                         0 * Metre});
     Length const semi_major_axis = (q1 - q2).Norm();
     period = 2 * π *
              Sqrt(Pow<3>(semi_major_axis) / (earth->gravitational_parameter() +
                                              moon->gravitational_parameter()));
     centre_of_mass =
-        Barycentre<Position<ICRFJ2000Equator>, Mass>(
+        Barycentre<Position<ICRS>, Mass>(
```
Comment (pleroy) [inline physics/ephemeris_test.cpp]: Fits on the previous line?

File: physics/ephemeris_test.cpp [hunk 27]
```diff
@@ -166,24 +166,24 @@ class EphemerisTest
...
-  std::vector<DegreesOfFreedom<ICRFJ2000Equator>> initial_state;
-  Position<ICRFJ2000Equator> centre_of_mass;
+  std::vector<DegreesOfFreedom<ICRS>> initial_state;
+  Position<ICRS> centre_of_mass;
   Time period;
   SetUpEarthMoonSystem(bodies, initial_state, centre_of_mass, period);
 
-  Ephemeris<ICRFJ2000Equator>
+  Ephemeris<ICRS>
       ephemeris(
```
Comment (pleroy) [inline physics/ephemeris_test.cpp]: Fits on the previous line?  (Many places.)

File: physics/ephemeris_test.cpp [hunk 28]
```diff
@@ -210,89 +210,89 @@ TEST_P(EphemerisTest, FlowWithAdaptiveStepSpecialCase) {
...
   MasslessBody probe;
-  DiscreteTrajectory<ICRFJ2000Equator> trajectory;
+  DiscreteTrajectory<ICRS> trajectory;
   trajectory.Append(t0_,
-                    DegreesOfFreedom<ICRFJ2000Equator>(
+                    DegreesOfFreedom<ICRS>(
                         earth_position +
-                            Displacement<ICRFJ2000Equator>(
+                            Displacement<ICRS>(
                                 {0 * Metre, distance, 0 * Metre}),
```
Comment (pleroy) [inline physics/ephemeris_test.cpp]: Fits on the previous line?

File: testing_utilities/solar_system_factory_test.cpp [hunk 29]
```diff
@@ -73,19 +73,19 @@ class SolarSystemFactoryTest : public testing::Test {
...
       std::string const& message) {
-    RelativeDegreesOfFreedom<ICRFJ2000Equator> const tertiary_secondary =
+    RelativeDegreesOfFreedom<ICRS> const tertiary_secondary =
         tertiary_dof - secondary_dof;
-    KeplerOrbit<ICRFJ2000Equator> orbit{
+    KeplerOrbit<ICRS> orbit{
         secondary_body, tertiary_body, tertiary_secondary, J2000};
-    Vector<Length, ICRFJ2000Equator> const& r =
+    Vector<Length, ICRS> const& r =
         tertiary_secondary.displacement();
```
Comment (pleroy) [inline testing_utilities/solar_system_factory_test.cpp]: Fits on the previous line?

File: astronomy/frames.hpp (hunk 1 above)
Comment (eggrobin) [reply astronomy/frames.hpp]: No, the question is whether this should be called BCRS instead of ICRS.

File: astronomy/frames.hpp (hunk 2 above)
Comment (eggrobin) [reply astronomy/frames.hpp]: http://articles.adsabs.harvard.edu/cgi-bin/nph-iarticle_query?db_key=AST&bibcode=1998A%26A...334.1123K&letter=.&classic=YES&defaultprint=YES&whole_paper=YES&page=1123&epage=1123&send=Send+PDF&filetype=.pdf

File: astronomy/mercury_perihelion_test.cpp (hunk 4 above)
Comment (eggrobin) [reply astronomy/mercury_perihelion_test.cpp]: No.

File: benchmarks/ephemeris.cpp (hunk 14 above)
Comment (eggrobin) [reply benchmarks/ephemeris.cpp]: No.

File: benchmarks/ephemeris.cpp (hunk 13 above)
Comment (eggrobin) [reply benchmarks/ephemeris.cpp]: No.

File: astronomy/mercury_perihelion_test.cpp (hunk 6 above)
Comment (eggrobin) [reply astronomy/mercury_perihelion_test.cpp]: No.

File: astronomy/mercury_perihelion_test.cpp (hunk 5 above)
Comment (eggrobin) [reply astronomy/mercury_perihelion_test.cpp]: No.

File: astronomy/frames.hpp [hunk 30]
```diff
@@ -19,57 +19,131 @@ using quantities::si::ArcMinute;
...
-using internal_frames::ICRFJ2000Equator;
-using internal_frames::ICRFJ200EquatorialToEcliptic;
-using internal_frames::SolarSystemBarycentreEcliptic;
-using internal_frames::SolarSystemBarycentreEquator;
+using internal_frames::GCRS;
+using internal_frames::ICRS;
+using internal_frames::ITRS;
 using internal_frames::Sky;
+using internal_frames::TerrestrialIntermediateReferenceSystem;
+using internal_frames::CelestialIntermediateReferenceSystem;
```
Comment (pleroy) [inline astronomy/frames.hpp]: Alphabetize this list.

File: benchmarks/ephemeris.cpp [hunk 31]
```diff
@@ -210,29 +213,40 @@ void EphemerisL4ProbeBenchmark(SolarSystemFactory::Accuracy const accuracy,
...
         at_спутник_1_launch->degrees_of_freedom(
             SolarSystemFactory::name(SolarSystemFactory::Earth));
-    Displacement<ICRFJ2000Ecliptic> const sun_earth_displacement =
-        ICRFJ200EquatorialToEcliptic(
+
+    // The BCRS, but with non-ICRS axes; the x axis is the ICRS x axis, the xy
+    // plane is the ecliptic realized by the obliquity given by the IAU in 1976
+    // (16th general assembly, resolution 10, commission 4, recommendation 1),
+    // identifying the ICRS xy plane with the mean equator of J2000.0.
+    struct Ecliptic;
```
Comment (pleroy) [inline benchmarks/ephemeris.cpp]: Could we move this declaration and the comment before the loop?

File: benchmarks/ephemeris.cpp [hunk 32]
```diff
@@ -210,29 +213,40 @@ void EphemerisL4ProbeBenchmark(SolarSystemFactory::Accuracy const accuracy,
...
+        from_barycentric(earth_degrees_of_freedom.velocity() -
+                         sun_degrees_of_freedom.velocity()));
+    Velocity<Ecliptic> const sun_l4_velocity = l4_rotation(sun_earth_velocity);
     trajectory.Append(at_спутник_1_launch->epoch(),
                       DegreesOfFreedom<Barycentric>(
                           sun_degrees_of_freedom.position() +
                               to_barycentric(
-                                  ICRFJ200EquatorialToEcliptic.Inverse()(
+                                  ecliptic_to_equatorial(
                                       sun_l4_displacement)),
```
Comment (pleroy) [inline benchmarks/ephemeris.cpp]: Surely this one fits on the previous line?  (Same 3 lines below.)

File: ksp_plugin_test/plugin_test.cpp (hunk 16 above)
Comment (pleroy) [reply ksp_plugin_test/plugin_test.cpp]: Still not fixed.

File: physics/barycentric_rotating_dynamic_frame_test.cpp [hunk 33]
```diff
@@ -219,26 +214,21 @@ TEST_F(BarycentricRotatingDynamicFrameTest, CoriolisAcceleration) {
...
-                                   0 * Metre / Second})};
-  DegreesOfFreedom<ICRFJ2000Equator> const small_dof =
-      {Displacement<ICRFJ2000Equator>({5 * Metre, 5 * Metre, 0 * Metre}) +
-           ICRFJ2000Equator::origin,
-       Velocity<ICRFJ2000Equator>({40 * Metre / Second,
-                                   -30 * Metre / Second,
-                                   0 * Metre / Second})};
-  DegreesOfFreedom<ICRFJ2000Equator> const barycentre_dof =
-      Barycentre<DegreesOfFreedom<ICRFJ2000Equator>, GravitationalParameter>(
+  DegreesOfFreedom<ICRS> const big_dof = {
```
Comment (pleroy) [inline physics/barycentric_rotating_dynamic_frame_test.cpp]: I am not in love with `{` on a line of its own.  Could we put it at the beginning of the next line?

File: physics/body_centred_body_direction_dynamic_frame_test.cpp [hunk 34]
```diff
@@ -202,16 +200,14 @@ TEST_F(BodyCentredBodyDirectionDynamicFrameTest, CoriolisAcceleration) {
...
-           ICRFJ2000Equator::origin,
-       Velocity<ICRFJ2000Equator>()};
-  DegreesOfFreedom<ICRFJ2000Equator> const small_dof =
-      {Displacement<ICRFJ2000Equator>({3 * Metre, 4 * Metre, 0 * Metre}) +
-           ICRFJ2000Equator::origin,
-       Velocity<ICRFJ2000Equator>({40 * Metre / Second,
-                                   -30 * Metre / Second,
-                                   0 * Metre / Second})};
+  DegreesOfFreedom<ICRS> const big_dof =
+      {Displacement<ICRS>({0 * Metre, 0 * Metre, 0 * Metre}) +
```
Comment (pleroy) [inline physics/body_centred_body_direction_dynamic_frame_test.cpp]: Here it's done well but not 5 lines below.

File: physics/body_test.cpp [hunk 35]
```diff
@@ -258,31 +258,28 @@ TEST_F(BodyTest, AllFrames) {
...
   TestRotatingBody<serialization::Frame::PluginTag,
-                  serialization::Frame::WORLD>();
+                   serialization::Frame::WORLD>();
   TestRotatingBody<serialization::Frame::PluginTag,
-                  serialization::Frame::WORLD_SUN>();
+                   serialization::Frame::WORLD_SUN>();
 
   TestRotatingBody<serialization::Frame::SolarSystemTag,
-                  serialization::Frame::ICRF_J2000_ECLIPTIC>();
+                   serialization::Frame::ICRF_J2000_ECLIPTIC>();
```
Comment (pleroy) [inline physics/body_test.cpp]: Do we still need this?

File: physics/ephemeris_test.cpp [hunk 36]
```diff
@@ -871,68 +822,68 @@ TEST_P(EphemerisTest, ComputeGravitationalAccelerationMassiveBody) {
...
                                         0 * AstronomicalUnit,
                                         1 * AstronomicalUnit});
   initial_state.emplace_back(q0, v);
   initial_state.emplace_back(q1, v);
   initial_state.emplace_back(q2, v);
   initial_state.emplace_back(q3, v);
 
-  Ephemeris<ICRFJ2000Equator>
+  Ephemeris<ICRS>
       ephemeris(
```
Comment (pleroy) [inline physics/ephemeris_test.cpp]: Would fit on the previous line.

File: physics/kepler_orbit_test.cpp [hunk 37]
```diff
@@ -491,54 +483,51 @@ TEST_F(KeplerOrbitTest, HyperbolicMeanAnomalyToTrueAnomaly) {
...
               Eq(*SimpleEllipse().argument_of_periapsis));
   EXPECT_THAT(*elements.longitude_of_periapsis,
               AlmostEquals(*SimpleEllipse().longitude_of_periapsis, 0));
 }
 
 #define CONSTRUCT_CONIC_FROM_TWO_ELEMENTS(element1, element2, reference)       \
-  \
-[&]() {                                                                        \
-    KeplerianElements<ICRFJ2000Equator> elements;                              \
+                                                                               \
```
Comment (pleroy) [inline physics/kepler_orbit_test.cpp]: Why the empty line.

File: physics/massive_body_body.hpp [hunk 38]
```diff
@@ -155,8 +155,10 @@ inline not_null<std::unique_ptr<MassiveBody>> MassiveBody::ReadFromMessage(
       using Tag = serialization::Frame::SolarSystemTag;
       if (enum_descriptor == google::protobuf::GetEnumDescriptor<Tag>()) {
         switch (static_cast<Tag>(enum_value_descriptor->number())) {
-          ROTATING_BODY_TAG_VALUE_CASE(ICRF_J2000_ECLIPTIC);
-          ROTATING_BODY_TAG_VALUE_CASE(ICRF_J2000_EQUATOR);
+          ROTATING_BODY_TAG_VALUE_CASE(ICRS);
+          ROTATING_BODY_TAG_VALUE_CASE(SKY);
+          ROTATING_BODY_TAG_VALUE_CASE(ITRS);
+          ROTATING_BODY_TAG_VALUE_CASE(GCRS);
```
Comment (pleroy) [inline physics/massive_body_body.hpp]: Alphabetize.

File: serialization/geometry.proto [hunk 39]
```diff
@@ -162,9 +162,12 @@ message Frame {
...
   // The frame tags used in the astronomical simulations.
   enum SolarSystemTag {
-    ICRF_J2000_ECLIPTIC = 1;
-    ICRF_J2000_EQUATOR = 2;
+    reserved 1;
+    reserved "ICRF_J2000_ECLIPTIC";
+    ICRS = 2;
     SKY = 3;
+    ITRS = 4;
+    GCRS = 5;
```
Comment (pleroy) [inline serialization/geometry.proto]: Alphabetize (nevermind the tags).  Move the reserved stuff to the end.

File: astronomy/frames.hpp [hunk 40]
```diff
@@ -151,8 +151,8 @@ using internal_frames::GCRS;
 using internal_frames::ICRS;
 using internal_frames::ITRS;
 using internal_frames::Sky;
-using internal_frames::TerrestrialIntermediateReferenceSystem;
 using internal_frames::CelestialIntermediateReferenceSystem;
+using internal_frames::TerrestrialIntermediateReferenceSystem;
```
Comment (pleroy) [inline astronomy/frames.hpp]: Alphabetize more.

File: ksp_plugin_test/plugin_integration_test.cpp [hunk 41]
```diff
@@ -109,9 +109,9 @@ class PluginIntegrationTest : public testing::Test {
                  satellite_initial_displacement_.Norm()) * unit_tangent;
   }
 
-  DegreesOfFreedom<Barycentric> ICRFToBarycentric(
+  DegreesOfFreedom<Barycentric> icrsToBarycentric(
```
Comment (pleroy) [inline ksp_plugin_test/plugin_integration_test.cpp]: Bad capitalization.

File: ksp_plugin_test/plugin_integration_test.cpp [hunk 42]
```diff
@@ -134,7 +134,7 @@ class PluginIntegrationTest : public testing::Test {
   }
 
   Identity<ICRS, Barycentric> ircf_to_barycentric_linear_;
```
Comment (pleroy) [inline ksp_plugin_test/plugin_integration_test.cpp]: `s/ircf/icrs/`

File: ksp_plugin_test/plugin_test.cpp (hunk 16 above)
Comment (pleroy) [reply ksp_plugin_test/plugin_test.cpp]: Still not fixed.  `s/icrf/icrs/`

**Pull request merged**